serrano/resources/query/__init__.py

```diff
@@ -0,0 +1,116 @@
+from django.conf.urls import patterns, url
+from django.views.decorators.cache import never_cache
+
+from avocado.models import DataQuery
+from serrano.resources import templates
+from serrano.resources.history import ObjectRevisionResource, \
+    ObjectRevisionsResource, RevisionsResource
+from serrano.resources.query.base import PublicQueriesResource, \
+    QueryResource, QueriesResource
+from serrano.resources.query.forks import QueryForksResource
+from serrano.resources.query.results import QueryResultsResource
+from serrano.resources.query.stats import QueryStatsResource
+
+single_resource = never_cache(QueryResource())
+active_resource = never_cache(QueriesResource())
+public_resource = never_cache(PublicQueriesResource())
+forks_resource = never_cache(QueryForksResource())
+stats_resource = never_cache(QueryStatsResource())
+results_resource = never_cache(QueryResultsResource())
+
+revisions_resource = never_cache(RevisionsResource(
+    object_model=DataQuery, object_model_template=templates.Query,
+    object_model_base_uri='serrano:queries'))
+revisions_for_object_resource = never_cache(ObjectRevisionsResource(
+    object_model=DataQuery, object_model_template=templates.Query,
+    object_model_base_uri='serrano:queries'))
+revision_for_object_resource = never_cache(ObjectRevisionResource(
+    object_model=DataQuery, object_model_template=templates.Query,
+    object_model_base_uri='serrano:queries'))
+
+
+# Resource endpoints
+urlpatterns = patterns(
+    '',
+
+    url(
+        r'^$',
+        active_resource,
+        name='active'
+    ),
+
+    # Endpoints for specific queries
+    url(
+        r'^public/$',
+        public_resource,
+        name='public'
+    ),
+
+    # Single queries
+    url(
+        r'^(?P<pk>\d+)/$',
+        single_resource,
+        name='single'
+    ),
+    url(
+        r'^session/$',
+        single_resource,
+        {'session': True},
+        name='session'
+    ),
+
+    # Endpoint for retrieving results of an existing query.
+    url(
+        r'^(?P<pk>\d+)/results/$',
+        results_resource,
+        name='results'
+    ),
+    url(
+        r'^(?P<pk>\d+)/results/(?P<page>\d+)/$',
+        results_resource,
+        name='results'
+    ),
+    url(
+        r'^(?P<pk>\d+)/results/(?P<page>\d+)\.\.\.(?P<stop_page>\d+)/$',
+        results_resource,
+        name='results'
+    ),
+
+    # Stats
+    url(
+        r'^(?P<pk>\d+)/stats/$',
+        stats_resource,
+        name='stats'
+    ),
+    url(
+        r'^session/stats/$',
+        stats_resource,
+        {'session': True},
+        name='stats'
+    ),
+
+    # Forks
+    # TODO add endpoint for session?
```

Comment:
Guess I misplaced this TODO in my shifting code between files. What I meant to ask is, is there any use for a `session/results/` URL? I didn't think it would prove useful but I wanted to note it in case it seemed useful to anyone else. Hopefully the above question makes much more(any?) sense.